webapp/components/topbar/dialogs/LaunchDialog.js

```diff
@@ -214,90 +225,131 @@ const useStyles = (theme) => ({
             name="radio-button"
           />
           <Box className="wrap">
-            <img src={value === 'experiment' ? experimentSelected : experimentUnselected} alt="completeExperiment" />
+            <img src={value === LAUNCH_MODAL.experimentState ? experimentSelected : experimentUnselected} alt="completeExperiment" />
             <Typography>Complete Experiment</Typography>
           </Box>
         </Typography>
       </Box>
 
-      <Accordion>
+      <Accordion expanded={expandConfiguration}>
         <AccordionSummary
           aria-label="Expand"
           aria-controls="additional-actions-content"
           id="additional-actions-header"
         >
           <Typography>
             <InfoIcon />
-            Run Configuration : Local Machine
-          </Typography>
-          <Typography component="span">
-            Edit
+            {`Run Configuration : ${LAUNCH_MODAL.defaultResource}`}
           </Typography>
+          <Button onClick={expandConfiguration ? (e) => handleConfigurationUpdate(e) : () => setExpandConfiguration(true)}>
+            { expandConfiguration ? 'Save' : 'Edit' }
+          </Button>
         </AccordionSummary>
         <AccordionDetails>
-          <FormControl variant="filled">
-            <InputLabel id="select-filled-label">Type</InputLabel>
-            <Select
-              labelId="select-filled-label"
-              id="select-filled-filled"
-              value="machine"
-              IconComponent={ExpandMoreIcon}
-            >
-              <MenuItem value="machine">Local Machine</MenuItem>
-            </Select>
-          </FormControl>
-          <TextField
-            variant="filled"
-            label="Field X"
-            fullWidth
-          />
-          <TextField
-            variant="filled"
-            label="Field Y"
-            fullWidth
-          />
+          { loading ? (
+            <Box className="primary-loader">
+              <CircularLoader />
+              Loading ...
+            </Box>
+          ) : (
+            <>
+              <FormControl variant="filled">
+                <InputLabel id="select-filled-label">Resources</InputLabel>
+                <Select
+                  labelId="select-filled-label"
+                  id="select-filled-filled"
+                  value="machine"
+                  IconComponent={ExpandMoreIcon}
+                >
+                  <MenuItem value="machine">{LAUNCH_MODAL.defaultResource}</MenuItem>
+                </Select>
+              </FormControl>
+              <FormControl variant="filled">
+                <InputLabel id="method">Method</InputLabel>
+                <Select
+                  labelId="method"
+                  id="method"
+                  value="method"
+                  IconComponent={ExpandMoreIcon}
+                >
+                  <MenuItem value="method">Bulletin</MenuItem>
+                </Select>
+              </FormControl>
+              <TextField
```

Comment:
use here `number` type and specify max and min attributes
min = 1 and max = X , we fetch X from backend, just set a dummy value